templates/profile/hathitrust/solr6/index-release.sh.erb

```diff
@@ -1,117 +1,108 @@
-#! /bin/bash
+#!/bin/bash
 
-TODAY=`date +%Y-%m-%d`
-#TODAY="2019-08-31"
+onoe () {
+  echo "${1}"
+  exit 1
+}
 
-BASE=/htsolr/<%= @solr_name %>
-SYMLINKBASE=/htsolr/serve
-SNAP=".snapshot/<%= @snapshot_name %>_${TODAY}"
-CORES="<%= @solr_cores.join(' ') %>"
+base=/htsolr/<%= @solr_name %>
+serve=/htsolr/serve
+snaps=( "${base}/.zfs/snapshot/htsolr-<%= @solr_name %>_"* )
+manual_snap="$1"
+if [[ -n "${manual_snap}" ]]; then
+  snap="${base}/.zfs/snapshot/${manual_snap}"
+  if ! [[ " ${array[@]} " =~ " ${snap} " ]]; then
```

Comment:
Could just check if the directory exists rather than the more complicated array thing:

```bash
if ! [[ -d "${snap}" ]]
```